### PR TITLE
Subclass VmScan job as ManageIQ::Providers::Microsoft::InfraManager::Scanning::Job

### PR DESCRIPTION
**Pull request by chessbyte**
State Machine diagram for this subclass of VmScan.

manageiq-providers-microsoft-infra_manager-scanning-job

### CODE REVIEW

**Comment by chessbyte** [comment]
@roliveri @jerryk55 if you have the environment to test SmartState for SCVMM with this and the [base](https://github.com/ManageIQ/manageiq/pull/19607) PR, it would be much appreciated.

File: app/models/manageiq/providers/microsoft/infra_manager/scanning/job.rb [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+      )
+    end
+  end
+
+  def before_scan
+    signal(:start_snapshot)
+  end
+
+  def call_snapshot_create
+    _log.info("Enter")
```
Comment (djberg96) [inline app/models/manageiq/providers/microsoft/infra_manager/scanning/job.rb]: I vote for a nicer message here. :)

File: app/models/manageiq/providers/microsoft/infra_manager/scanning/job.rb [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+      if vm.ext_management_system
+        _log.info("Deleting snapshot: reference: [#{mor}]")
+        begin
+          delete_snapshot(mor, vm)
+        rescue => err
+          _log.error(err.to_s)
+          return
+        rescue Timeout::Error
+          msg = "Request to delete snapshot timed out"
+          _log.error(msg)
```
Comment (djberg96) [inline app/models/manageiq/providers/microsoft/infra_manager/scanning/job.rb]: Same issue here as I mentioned in the other one. Without a return, won't this look like it succeeded?

File: app/models/manageiq/providers/microsoft/infra_manager/scanning/job.rb (hunk 1 above)
Comment (chessbyte) [reply app/models/manageiq/providers/microsoft/infra_manager/scanning/job.rb]: @djberg96 while I agree, I would do that in a separate PR.  I am just refactoring here - trying to change the least amount possible and ensure everything still works.

File: app/models/manageiq/providers/microsoft/infra_manager/scanning/job.rb (hunk 2 above)
Comment (chessbyte) [reply app/models/manageiq/providers/microsoft/infra_manager/scanning/job.rb]: @djberg96 I wonder how this worked prior to this refactoring.

**Pull request merged**